example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,363 @@
+      focalPoint: globalFocalPoint,
+      allowHorizontalArrow: allowHorizontalArrow,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant RenderPopover renderObject) {
+    super.updateRenderObject(context, renderObject);
+    renderObject
+      ..radius = radius
+      ..arrowBaseWidth = arrowBaseWidth
+      ..arrowLength = arrowLength
+      ..padding = padding
+      ..focalPoint = globalFocalPoint
+      ..backgroundColor = backgroundColor
+      ..allowHorizontalArrow = allowHorizontalArrow;
+  }
+}
+
+class RenderPopover extends RenderShiftedBox {
+  RenderPopover({
+    required Radius radius,
+    required double arrowWidth,
+    required double arrowLength,
+    required Color backgroundColor,
+    required Offset focalPoint,
+    bool allowHorizontalArrow = true,
+    EdgeInsets? padding,
+    RenderBox? child,
+  })  : _radius = radius,
+        _arrowBaseWidth = arrowWidth,
+        _arrowLength = arrowLength,
+        _padding = padding,
+        _backgroundColor = backgroundColor,
+        _backgroundPaint = Paint()..color = backgroundColor,
+        _focalPoint = focalPoint,
+        _allowHorizontalArrow = allowHorizontalArrow,
+        super(child);
+
+  Radius _radius;
+  Radius get radius => _radius;
+  set radius(Radius value) {
+    if (_radius != value) {
+      _radius = value;
+      markNeedsLayout();
+    }
+  }
+
+  double _arrowBaseWidth;
+  double get arrowBaseWidth => _arrowBaseWidth;
+  set arrowBaseWidth(double value) {
+    if (_arrowBaseWidth != value) {
+      _arrowBaseWidth = value;
+      markNeedsLayout();
+    }
+  }
+
+  double _arrowLength;
+  double get arrowLength => _arrowLength;
+  set arrowLength(double value) {
+    if (_arrowLength != value) {
+      _arrowLength = value;
+      markNeedsLayout();
+    }
+  }
+
+  Offset _focalPoint;
+  Offset get focalPoint => _focalPoint;
+  set focalPoint(Offset value) {
+    if (_focalPoint != value) {
+      _focalPoint = value;
+      markNeedsLayout();
+    }
+  }
+
+  EdgeInsets? _padding;
+  EdgeInsets? get padding => _padding;
+  set padding(EdgeInsets? value) {
+    if (_padding != value) {
+      _padding = value;
+      markNeedsLayout();
+    }
+  }
+
+  Color _backgroundColor;
+  Color get backgroundColor => _backgroundColor;
+  set backgroundColor(Color value) {
+    if (value != _backgroundColor) {
+      _backgroundColor = value;
+      _backgroundPaint = Paint()..color = _backgroundColor;
+      markNeedsPaint();
+    }
+  }
+
+  bool _allowHorizontalArrow;
+  bool get allowHorizontalArrow => _allowHorizontalArrow;
+  set allowHorizontalArrow(bool value) {
+    if (value != _allowHorizontalArrow) {
+      _allowHorizontalArrow = value;
+      markNeedsLayout();
+    }
+  }
+
+  late Paint _backgroundPaint;
+
+  @override
+  void performLayout() {
+    // We need to know our offset in order to calculate the arrow direction
+    // and we only know our offset at paint phase.
```

Comment:
I should rephrase this. What I mean is that we don't know our offset until the paint phase.

Reply:
What should we do when we get infinity constraints?

Reply:
I think what we should do is check if the child menu has an intrinsic size. If the child does have an intrinsic size, then let the child be as large as it wants. If the child doesn't have an intrinsic size, e.g., a menu with a width of `double.infinity`, then we should pass the screen size as constraints.

Reply:
Actually, that's probably a bad idea, because it wouldn't allow for the iOS toolbar with buttons that exceed the width of the screen. I guess we should impose screen bounds all the time.